applications/tari_base_node/src/builder.rs

```diff
@@ -467,12 +467,45 @@ where
 
     // Set the base node for the wallet to the 'local' base node
     let base_node_public_key = base_node_comms.node_identity().public_key().clone();
-    wallet_handles
+    let mut transaction_service_handle = wallet_handles
         .get_handle::<TransactionServiceHandle>()
-        .expect("TransactionService is not registered")
+        .expect("TransactionService is not registered");
+    transaction_service_handle
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
+    transaction_service_handle
```

Comment:
I think this should also be delayed until the initial sync has been confirmed, which should be the main priority at startup. For example, **has the Tx been mined** is dependent on an up to date blockchain.

Reply:
So with this refactor all this does is set the base node pub key, the protocols are only started when the explicit functions are called which is done in the task that waits for the sync to complete

Reply:
Please see comment below